### ISSUE TITLE
Add support for `export` keyword

### ISSUE THREAD

**Issue by chenasraf** [dotenvx/dotenvx#355]
## Description of the issue

It's common for developers to have a `.env` file which is a shell script, so they can easily source them manually if the need arises.
Often they will use `export` for a variable for that purpose as well.

Take the following file as example:

```.env
#!/usr/bin/env bash
export KEY=value
```

It is a valid and loadable env file for many tools such as `direnv`, `dotenv` (NPM), etc, and also supports direct source

However, it does not play well with `dotenvx`. Trying to migrate and encrypt the values does not result in the desired output.

```sh
❯ dotenvx encrypt
✔ encrypted (.env)
✔ key added to .env.keys (DOTENV_PRIVATE_KEY)
ℹ add .env.keys to .gitignore: [echo ".env.keys" >> .gitignore]
ℹ run [DOTENV_PRIVATE_KEY='05baf46b1de94218e746b410765be902ed29921ca62114cedb65fa3b8cc31990' dotenvx run -- yourcommand] to test decryption locally

❯ cat .env
#/-------------------[DOTENV_PUBLIC_KEY]--------------------/
#/            public-key encryption for .env files          /
#/       [how it works](https://dotenvx.com/encryption)     /
#/----------------------------------------------------------/
DOTENV_PUBLIC_KEY="0243547f7fa8eb4dc04382166c1ecc255d90d6e336bde9a667b86ef175b7f62ebf"

# .env
#!/usr/bin/env bash
export KEY=value

❯ cat .env.keys
#/------------------!DOTENV_PRIVATE_KEYS!-------------------/
#/ private decryption keys. DO NOT commit to source control /
#/     [how it works](https://dotenvx.com/encryption)       /
#/----------------------------------------------------------/

# .env
DOTENV_PRIVATE_KEY="05baf46b1de94218e746b410765be902ed29921ca62114cedb65fa3b8cc31990"
```

1. The keys were properly created and added
2. The value was **not encrypted** at all

Using `dotenvx set` also doesn't resolve this:

```sh
❯ dotenvx set KEY value
✔ set KEY with encryption (.env)

❯ cat .env
#/-------------------[DOTENV_PUBLIC_KEY]--------------------/
#/            public-key encryption for .env files          /
#/       [how it works](https://dotenvx.com/encryption)     /
#/----------------------------------------------------------/
DOTENV_PUBLIC_KEY="0243547f7fa8eb4dc04382166c1ecc255d90d6e336bde9a667b86ef175b7f62ebf"

# .env
#!/usr/bin/env bash
export KEY=value
```

## Expected behavior

Since `export` is probably not relevant anymore as sourcing directly won't work anyway, the `export` keyword doesn't have to remain in the final output.

However, encrypting a file or setting a value should still work out of the box, to reduce friction by a lot. (Took me a solid 20 minutes to realize this was the source of the issue, I suspect not everyone will be able to in a reasonable time)

I expect the output to probably be:

```sh
❯ cat .env
#/-------------------[DOTENV_PUBLIC_KEY]--------------------/
#/            public-key encryption for .env files          /
#/       [how it works](https://dotenvx.com/encryption)     /
#/----------------------------------------------------------/
DOTENV_PUBLIC_KEY="0243547f7fa8eb4dc04382166c1ecc255d90d6e336bde9a667b86ef175b7f62ebf"

# .env
KEY="encrypted:........."
```

Also note that the shebang also moves to an incorrect place, it should either be removed or remain at the top.

## Proposal

This requires 2 things to be mainly done:

1. Find the correct lines even if they are prefixed with `"export "`
2. Ignore the `export` keyword when parsing the key name before encrypting

As for the shebang, I don't think it's as high of a priority, but could also be handled better.

What is the stance on this from the author(s)?

I might be able to make a PR myself but before I work on it I would like to know if it's something you guys want to resolve.

If it's not to be resolved, maybe the docs could be updated to reflect this caveat. The simplest solution on the user's end is to simply remove all `export ` prefixes before encrypting, but it should be mentioned somewhere (if it was, I apologize - but I couldn't find it)

Thanks for a truly great package either way :)

**Comment by chenasraf** [comment]
Just noticed these related issue: 

- https://github.com/motdotla/dotenv/issues/213
- https://github.com/motdotla/dotenv/issues/521

Is there any progress done in the subject, or anywhere to track progress? Would love to see this implemented.

**Comment by motdotla** [comment]
dotenv supports export these days: https://github.com/motdotla/dotenv/blob/master/lib/main.js#L9

the problem here is the encryption logic in dotenvx is not smart enough to recognize the export KEY. i'll patch up.

**Comment by motdotla** [comment]
> Thanks for a truly great package either way :)

It's been my pleasure. 🎩

**Comment by motdotla** [comment]
fix going out now as lucky number 11 11: `v1.11.1` https://github.com/dotenvx/dotenvx/pull/357

it also preserves shebangs now.

**Comment by chenasraf** [comment]
Wonderful! Thank you :)
